docs/cqrs/index.md

```diff
@@ -0,0 +1,21 @@
+# CQRS
+
+**Command-Query Responsibility Segregation** is a pattern that promotes hard split between parts of code (classes/objects) that read data ([query]) and the ones that modify and write the data ([command]). It also separates the payload (e.g. object id, user name) from the code that executes the query/command.
+
+This library went a step further and used commands and queries as client-facing API. It leverages `HttpContext` to define the environment in which the command or query handler operates. This design choice enables data retrieval from side channels such as cookies or HTTP headers.
```

Comment:
Mentioning `HttpContext` here is too early - it should describe the `why`, not the code-level details.